numpy_financial/_financial.py

```diff
@@ -601,10 +599,17 @@ def pv(rate, nper, pmt, fv=0, when='end'):
     """
     when = _convert_when(when)
     (rate, nper, pmt, fv, when) = map(np.asarray, [rate, nper, pmt, fv, when])
-    temp = (1 + rate) ** nper
-    fact = np.where(rate == 0, nper, (1 + rate * when) * (temp - 1) / rate)
-    return -(fv + pmt * fact) / temp
+    
+    try:
+        temp = (1 + rate) ** nper
+        fact = np.where(rate == 0, nper, (1 + rate * when) * (temp - 1) / rate)
+        return -(fv + pmt * fact) / temp
+
+    except (InvalidOperation, TypeError, ValueError, DivisionByZero, Overflow) as e:
```

Comment:
Returning `-0.0` may be controversial. In certain cases, `numpy` will return `-0.0` with a runtime warning.

Reply:
No longer applicable.